### DISCUSSION TITLE
Result webpart wont render when {{ellipsis}} handlebar is used

### DISCUSSION THREAD

**Discussion by bckrma** [microsoft-search/pnp-modern-search#2775] [Q&A] [answered]
**Version used**
Ex: 4.8

**Describe the bug**
The PnP search result webpart wont render if the {{ellipsis}} handlebar is used in the result template.

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Add PnP - Search Result webpart to site and adjust query text to display some items
2. Insert `{{ellipsis "foo bar baz", 7}}` into the result template
3. Refresh the site, only the webpart title will be rendered

**Expected behavior**
`{{ellipsis "foo bar baz", 7}}` should truncate the string `"foo bar baz"` to `'foo bar…'`
https://github.com/helpers/handlebars-helpers#ellipsis

**Screenshots**
result webpart without {{ellipsis}} handle bar
![bugreport_pnp_handlebar1](https://user-images.githubusercontent.com/48834093/219008205-ed7e4d73-2314-4862-a025-ed6a0a3a5469.png)

result webpart with {{ellipsis}} handlebar
![bugreport_pnp_handlebar3](https://user-images.githubusercontent.com/48834093/219008295-ced9020e-485d-4ba5-b94e-f2fdce96ea07.png)

Microsoft Edge Developer Tools - without {{ellipsis}} handlebar
![bugreport_pnp_handlebar4](https://user-images.githubusercontent.com/48834093/219008494-dc5088a5-6dd5-4133-8050-b6f94707b736.png)

Microsoft Edge Developer Tools with {{ellipsis}} handlebar
![bugreport_pnp_handlebar5](https://user-images.githubusercontent.com/48834093/219008934-80441315-ee6a-4024-a5dd-ca289a63e284.png)

**Desktop (please complete the following information):**
 - Browser: Microsoft Edge ( 110.0.1587.46 64-Bit)

**Additional context**
<details>
<summary>Result template from the example above</summary>
`<content id="data-content">

    <style>

       /* Insert your CSS overrides here */
       .example-themePrimary a {
            color: {{@root.theme.palette.themePrimary}};
        }

        .icon {
            width: 20px;
            height: 16px;
        }

        ul.template--custom {
            list-style: none;
            padding-left: 5px;
        }

        ul.template--custom li {
            display: flex;
            padding: 8px;
        }

    </style>

    <div class="template">

        {{#if @root.properties.showSelectedFilters}}
            <pnp-selectedfilters
                data-filters="{{JSONstringify filters.selectedFilters 2}}"
                data-filters-configuration="{{JSONstringify filters.filtersConfiguration 2}}"
                data-instance-id="{{filters.instanceId}}"
                data-operator="{{filters.filterOperator}}"
                data-theme-variant="{{JSONstringify @root.theme}}"
            ></pnp-selectedfilters>
        {{/if}}

        <div class="template--header">
            {{#if @root.properties.showResultsCount}}
                <div class="template--resultCount">
                    <label class="ms-fontWeight-semibold">{{getCountMessage @root.data.totalItemsCount @root.inputQueryText}}</label>
                </div>
            {{/if}}

            <div class="template--sort">
                <pnp-sortfield
                    data-fields="{{JSONstringify @root.properties.dataSourceProperties.sortList}}"
                    data-default-selected-field="{{sort.selectedSortFieldName}}"
                    data-default-direction="{{sort.selectedSortDirection}}"
                    data-theme-variant="{{JSONstringify @root.theme}}">
                </pnp-sortfield>
            </div>
        </div>

        <ul class="template--custom">
            {{#each data.items as |item|}}

                <pnp-select
                    data-enabled="{{@root.properties.itemSelectionProps.allowItemSelection}}"
                    data-index="{{@index}}"
                    data-is-selected="{{isItemSelected @root.selectedKeys @index}}">

                    <template id="content">

                        {{#> resultTypes item=item}}
                            {{!-- The block below will be used as default item template if no result types matched --}}
                             {{ellipsis "foo bar baz", 7}}
                            <li>
                                <pnp-iconfile class="icon" data-extension="{{slot item @root.slots.FileType}}" data-theme-variant="{{JSONstringify @root.theme}}"></pnp-iconfile>
                                <span class="example-themePrimary"><a href="{{slot item @root.slots.Path}}">{{slot item @root.slots.Title}}</a></span>
                            </li>
                        {{/resultTypes}}
                    </template>

                </pnp-select>

            {{/each}}
        </ul>
    </div>

    {{#if @root.properties.paging.showPaging}}
        <pnp-pagination
            data-total-items="{{@root.data.totalItemsCount}}"
            data-hide-first-last-pages="{{@root.properties.paging.hideFirstLastPages}}"
            data-hide-disabled="{{@root.properties.paging.hideDisabled}}"
            data-hide-navigation="{{@root.properties.paging.hideNavigation}}"
            data-range="{{@root.properties.paging.pagingRange}}"
            data-items-count-per-page="{{@root.properties.paging.itemsCountPerPage}}"
            data-current-page-number="{{@root.paging.currentPageNumber}}"
        >
        </pnp-pagination>
    {{/if}}

</content>

<content id="placeholder-content">

    <style>

        .placeholder .icon {
            width: 20px;
            height: 16px;
            margin-right: 5px;
        }

        .placeholder ul {
            list-style: none;
        }

        .placeholder ul li {
            display: flex;
            align-items: center;
            padding: 8px;
        }

    </style>

    <div class="placeholder">
        <ul>
            {{#times @root.properties.paging.itemsCountPerPage}}
                <li>
                    <div class="icon placeholder--shimmer"></div>
                    <span class="placeholder--shimmer placeholder--line" style="width: 60%"></span>
                </li>
            {{/times}}
        </ul>
    </div>
</content>`
</details>

**Comment by wobba** [comment, marked as answer]
Syntax is `{{ellipsis "foo bar baz" 7}}` without the comma, which is normal handlebar param syntax and the docs are wrong in their page.

**Reply by bckrma** [reply]
Oh okay, that was simpler than expected.
Thank you very much for the fast reply!
